src/components/Quiz/Quiz.js

```diff
@@ -1,28 +1,118 @@
 import './Quiz.css';
 import { NavLink } from 'react-router-dom/cjs/react-router-dom.min';
+import getCocktails from '../../api-calls';
+import { useState } from 'react';
+import listIngredients from '../../helper-functions';
 
 function Quiz() {
-    const spirits = ['vodka', 'gin', 'rum', 'whiskey', 'tequila', 'scotch', 'cordial', 'sparkling', 'non-alcoholic']
-    const ingredients = ['lime', 'lemon', 'coffee', 'soda', 'cranberry', 'orange', 'grapefruit', 'bitters', 'ginger']
-    const glassware = ['rocks', 'flute', 'hurricane', 'coupe', 'highball', 'shot', 'wine', 'martini']
+
+    const [cocktailResults, setCocktailResults] = useState([]);
+    const [questionNum, setQuestionNum] = useState(0);
+    const [quizError, setQuizError] = useState('');
+    const [cocktail, setCocktail] = useState({});
+
+    const spirits = ['Vodka', 'Gin', 'Rum', 'Bourbon', 'Tequila', 'Scotch'];
+    const ingredients = ['Lime', 'Lemon', 'Coffee', 'Cranberry_juice', 'Orange_juice', 'Grapefruit_juice', 'Bitters', 'Ginger'];
+    const glassware = ['Cocktail_glass', 'Champagne_flute', 'Hurricane_glass', 'Whiskey_sour_glass', 'Highball_glass', 'Shot_glass', 'Collins_glass', 'Martini_glass'];
 
     const makeButtons = (category) => {
-        return category.map(item => <button>{item}</button>)
+        return category.map((item, index) => <button name={item} onClick={(event) => fetchSelection(event.target.name)} key={index}>{item}</button>);
+    }
+
+    const fetchSelection = (selection) => {
+        let path;
+        if(questionNum < 2) {
+            path = 'filter.php?i=' + selection;
+        } else {
+            path = 'filter.php?g=' + selection;
+        }
+        const promise = getCocktails(path);
+        promise.then(data => {
+            if (typeof data === 'string' || data instanceof String) {
+                setQuizError(data);
+            } else {
+                filterCocktails(data.drinks);
+            }
+        });
+    }
+
+    const filterCocktails = (cocktails) => {
+        if(questionNum === 0) {
```

Comment:
It might slightly improve readability to have this conditional be a part of fetchSelection - unless there's a reason for it to be down here, my brain reads line 24 and thinks "ok we're checking how many questions are left" and then line 40 seems like the natural `else` to that statement.  I could be wrong assuming that that's a doable refactor - at a glance I didn't see anywhere else that uses filterCocktails but I have only been looking for a few minutes.

Reply:
The conditional in fetchSelection checks whether we are looking for an ingredient or glassware as they are different end points in the api. The conditional in filterCocktails looks if it's the first question or not to reset the array of cocktailResults. If it's not the first question then a comparison happens between the most recent fetch and the current array. Its a weird chain of events.... 🙃